syntax = "proto3";  // 声明了protobuf的版本

package fixbug; // 声明了代码所在的包（对于C++来说是namespace）

// 定义下面的选项，表示生成service服务类和rpc方法描述，默认不生成。
option cc_generic_services = true;

// 错误类型
message ResultCode
{
    int32 errcode=1;
    bytes errmsg = 2;
}

message GetFriendsListRequest
{
    uint32 userid=1;
}

message GetFriendsListResponse
{
    ResultCode result = 1;
    repeated bytes friends = 2;
}


//在protobuf里面怎么定义描述rpc方法的类型 - service
service FriendServiceRpc
{
    rpc GetFriendsList(GetFriendsListRequest) returns(GetFriendsListResponse);
}